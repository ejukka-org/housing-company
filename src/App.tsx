import React from "react";
import "./App.css";

import { BrowserRouter as Router, Route } from "react-router-dom";

import HomePage from "./pages/home/home.page";
import Header from "./components/header/header.component";
import AboutPage from "./pages/about/about.page";
import SignIn from "./components/sign-in/sign-in.component";
import "bootstrap/dist/css/bootstrap.min.css";
import ApartmentsPage from "./pages/apartments/apartments.page";
import HistoryPage from "./pages/history/history.page";

import { Provider, useDispatch } from "react-redux";

import { checkIsUserLoggedIn } from "./redux/user.action";
import reduxStore from "./redux/store";
import { PersistGate } from "redux-persist/integration/react";

const App = () => {
  const dispatch = useDispatch();
  dispatch(checkIsUserLoggedIn());
  const { store, persistor } = reduxStore;

  return (
    <PersistGate persistor={persistor}>
      <Provider store={store}>
        <div>
          <Router>
            <Header />
            <Route exact path="/" component={HomePage} />
            <Route exact path="/history" component={HistoryPage} />
            <Route exact path="/about" component={AboutPage} />
            <Route exact path="/apartments" component={ApartmentsPage} />
            <Route exact path="/signin" component={SignIn} />
          </Router>
        </div>
      </Provider>
    </PersistGate>
  );
};

export default App;
