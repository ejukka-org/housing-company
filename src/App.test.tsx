import App from "./App";
import { act } from "react-dom/test-utils";
import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import reduxStore from "./redux/store";
import pretty from "pretty";

let container: any;

beforeEach(() => {
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

it("renders without crashing", () => {
  const store = reduxStore.store;
  act(() => {
    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>,
      container
    );
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
