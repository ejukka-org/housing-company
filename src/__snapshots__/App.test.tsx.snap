// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
"<div>
  <div class=\\"container\\">
    <nav class=\\"navbar navbar-expand-lg navbar-dark bg-dark\\"><a href=\\"/\\" class=\\"navbar-brand\\">Housing Company</a><button aria-controls=\\"responsive-navbar-nav\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button>
      <div class=\\"navbar-collapse collapse\\" id=\\"responsive-navbar-nav\\">
        <div class=\\"mr-auto navbar-nav\\"><a href=\\"/apartments\\" data-rb-event-key=\\"/apartments\\" class=\\"nav-link\\">admin header text1</a><a href=\\"/history\\" data-rb-event-key=\\"/history\\" class=\\"nav-link\\">admin header text2</a>
          <div class=\\"dropdown nav-item\\"><a aria-haspopup=\\"true\\" aria-expanded=\\"false\\" href=\\"#\\" class=\\"dropdown-toggle nav-link\\" role=\\"button\\">admin header text3</a></div>
        </div>
        <div class=\\"navbar-nav\\"><a href=\\"/about\\" data-rb-event-key=\\"/about\\" class=\\"nav-link\\">admin header text4</a><button type=\\"button\\" class=\\"btn btn-primary\\">signin</button></div>
      </div>
    </nav>
  </div>
  <div class=\\"container\\">
    <div class=\\"container\\">
      <div class=\\"sc-bdVaJa eskUMO\\"><img style=\\"background-image: url();\\" class=\\"background-image\\">
        <div class=\\"sc-bwzfXH kXomTW\\">
          <h1 class=\\"sc-htpNat hgjgiu\\">MENU TITLE1</h1><span class=\\"sc-bxivhb fzOaMp\\">undefined</span>
        </div>
      </div>
    </div>
    <div class=\\"container\\">
      <div class=\\"sc-bdVaJa eskUMO\\"><img style=\\"background-image: url();\\" class=\\"background-image\\">
        <div class=\\"sc-bwzfXH kXomTW\\">
          <h1 class=\\"sc-htpNat hgjgiu\\">MENU TITLE2</h1><span class=\\"sc-bxivhb fzOaMp\\">undefined</span>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
